common/src/Assets/Palette.cpp

```diff
@@ -88,5 +124,51 @@ namespace TrenchBroom {
         bool Palette::initialized() const {
             return m_data.get() != nullptr;
         }
+
+        bool Palette::indexedToRgba(IO::BufferedReader& reader, const size_t pixelCount, TextureBuffer& rgbaImage, const PaletteTransparency transparency, Color& averageColor) const {
```

Comment:
Why do you require a buffered reader here?

Reply:
Just so I can implement this in terms of pointer access. The previous implementation was reading pixels one byte at a time (with at least 1 virtual function call into Reader) which is a no-go in terms of performance.

Reply:
Ah, I see. I wonder if it wouldn't be better to just pass in the raw data in that case?

Reply:
I started doing that, and there was more "spam" change at all of the `indexedToRgba` call sites (iirc there are 5).
This approach keeps the exception handling the same. We could always do that later.

Reply:
Ok!